code/game/machinery/computer/camera.dm

```diff
@@ -5,153 +5,175 @@
 	icon_keyboard = "security_key"
 	circuit = /obj/item/circuitboard/computer/security
 	light_color = LIGHT_COLOR_RED
+	ui_x = 870
+	ui_y = 708
 
-	var/last_pic = 1
 	var/list/network = list("ss13")
-	var/list/watchers = list() //who's using the console, associated with the camera they're on.
-	var/long_ranged = FALSE
+	var/obj/machinery/camera/active_camera
+	var/list/concurrent_users = list()
+
+	// Stuff needed to render the map
+	var/map_name
+	var/const/default_map_size = 15
+	var/obj/screen/cam_screen
+	var/obj/screen/plane_master/lighting/cam_plane_master
+	var/obj/screen/background/cam_background
 
 /obj/machinery/computer/security/Initialize()
 	. = ..()
+	// Map name has to start and end with an A-Z character,
+	// and definitely NOT with a square bracket or even a number.
+	// I wasted 6 hours on this. :agony:
+	map_name = "camera_console_[REF(src)]_map"
+	// Convert networks to lowercase
 	for(var/i in network)
 		network -= i
 		network += lowertext(i)
+	// Initialize map objects
+	cam_screen = new
+	cam_screen.name = "screen"
+	cam_screen.assigned_map = map_name
+	cam_screen.del_on_map_removal = FALSE
+	cam_screen.screen_loc = "[map_name]:1,1"
+	cam_plane_master = new
+	cam_plane_master.name = "plane_master"
+	cam_plane_master.assigned_map = map_name
+	cam_plane_master.del_on_map_removal = FALSE
+	cam_plane_master.screen_loc = "[map_name]:CENTER"
+	cam_background = new
+	cam_background.assigned_map = map_name
+	cam_background.del_on_map_removal = FALSE
+
+/obj/machinery/computer/security/Destroy()
+	. = ..()
+	qdel(cam_screen)
+	qdel(cam_plane_master)
+	qdel(cam_background)

```

Comment:
```suggestion
/obj/machinery/computer/security/Destroy()
	qdel(cam_screen)
	qdel(cam_plane_master)
	qdel(cam_background)
	return ..()
```
Prefer to perform the operations first on `Destroy()`, as we don't have guarantees on what will be broken down the line.